### DISCUSSION TITLE
Copy / paste shortcuts of files in navigator view ?

### DISCUSSION THREAD

**Discussion by ced75** [dbeaver/dbeaver#561] [Q&A] [answered]
Hi,

I have configured the new Navigator view and I would like to copy/paste files from one directory to another one, using ctrl+c / ctrl+v shortcuts.
This bindings are not configured by default.
I tried to add them to the Copy/Paste commands in my preferences, and I choosed "Navigation context" in "When:" box.
But it doesn't seem to work.

How can I configure these shortcuts for these two action in the Navigator view?

Thanks for your help.
Regards.

**Comment by serge-rider** [comment]
What DBeaver version do you have?
ctrlc/ctrl+v (as well as copy/paste commands from context/main menus) works fine in my test environment.
If you have the rcent version then maybe there is some error happen during copy/paste - please check  Error Log view.

**Comment by ced75** [comment]
DBeaver 3.7.0
Copy/Paste commands work fine from context menu. The shortcuts work fine in "Project" view and also appear in the context menu:
![image](https://cloud.githubusercontent.com/assets/12892581/16451890/ed2cb9b0-3e05-11e6-99d6-7ac9c1a10c13.png) ![image](https://cloud.githubusercontent.com/assets/12892581/16451895/f2a087a0-3e05-11e6-9af3-3b7eee9302a6.png)

But I do not have the usual shortcuts in context menu in "Navigator" view:
![image](https://cloud.githubusercontent.com/assets/12892581/16451910/068b0894-3e06-11e6-8454-3514c61bcbd6.png) ![image](https://cloud.githubusercontent.com/assets/12892581/16451924/17044dfc-3e06-11e6-8c78-8f0d9aa4e09f.png)

I did not find how customize copy/paste in this view.

**Comment by serge-rider** [comment]
Can't reproduce.
I found a bug with pasting external resources in the Project Explorer view. And I've fixed it.
But your problem is different (you can't paste file copied from DBeaver "Project Explorer" in standard Eclipse "Navigator" view. I can easily do it in 3.7 and 3.6.8+.

What's your Eclipse version?
Can you recheck DBeaver plugins versions?

**Comment by ced75** [comment]
OK, I think I forgot to precise an important detail: I am not using DBeaver as an Eclipse plugin. I am using it in standalone version.
The Navigator view has just been added in the standalone version since v 3.7.0.
And there, in the standalone version, the shortcuts do not appear.

Cedric

**Comment by serge-rider** [comment]
That's odd. Still can't reproduce that, "paste" is enabled in Navigator.
![navigator-paste](https://cloud.githubusercontent.com/assets/6398845/16566663/3c04936a-421f-11e6-8abe-1b861a882ddc.png)

BTW why do you use "Navigator" view? You can paste scripts right in Project Explorer (ctrl+v).

**Comment by ced75** [comment]
Yes, I also can copy and paste with right clic, but the shorcuts (ctrl+c, ctrl+v) do not work in this view.
I use preferencially navigator because of file type filters (my SQL scripts are in folders with several files of different types).

**Comment by serge-rider** [comment, marked as answer]
You can use resource filters in Project Explorer view as well (alt+enter on folder and edit resource filters).

**Comment by ced75** [comment]
OK, thanks for the tip.
